smart-contract/assembly/contracts/internals/fileInit.ts

```diff
@@ -0,0 +1,57 @@
+import { sha256, Storage } from '@massalabs/massa-as-sdk';
+import { _pushFileLocation } from './location';
+import { fileMetadataLocationKey } from './storageKeys/metadataKeys';
+import { Metadata } from '../serializable/Metadata';
+import { _editFileMetadata } from './metadata';
+import {
+  _getTotalChunk,
+  _removeChunksRange,
+  _setFileChunkCount,
+} from './chunks';
+import { stringToBytes } from '@massalabs/as-types';
+
+/**
+ * Initializes a file in the smart contract storage.
+ * @param location - The location of the file.
+ * @param hashLocation - The hash of the file location.
+ * @param totalChunk - The total number of chunks for the file.
+ * @param metadata - An array of Metadata objects for the file.
+ * @throws If the total chunk count is 0 or less.
+ */
+export function _fileInit(
+  location: string,
+  hashLocation: StaticArray<u8>,
+  totalChunk: u32,
+  metadata: Metadata[],
+): void {
+  assert(totalChunk > 0, 'Total chunk must be greater than 0');
+
+  // Add file location if it doesn't exist
+  if (!Storage.has(fileMetadataLocationKey(hashLocation))) {
+    assert(
+      sha256(stringToBytes(location)).toString() == hashLocation.toString(),
```

Comment:
since we already compute the hash to perform this check, might as well not ask the user to compute it, and simply ask the user to pass the raw location when calling the contract